test/DynamoCoreWpfTests/PackageManager/PackageManagerUITests.cs

```diff
@@ -1262,8 +1262,12 @@ public void PackageManagerDownloadsBeforeInstalling()
                 .Returns(MessageBoxResult.OK);
             MessageBoxService.OverrideMessageBoxDuringTests(dlgMock.Object);
 
-            //actually perform the download & install operations
-            pmVmMock.Object.ExecutePackageDownload(id, pkgVer, "");
+            Task.Run(() => {
+                // This test runs on a single thread (using DispatcherSyncronizationContext)
+                // We need to run the async ExecutePackageDownload function in a separate thread so that the Thread.Sleep does not mess up the other awaiting tasks.
+                //actually perform the download & install operations
+                pmVmMock.Object.ExecutePackageDownload(id, pkgVer, "");
+            }).Wait();
```

Comment:
Task.Run works here because there is no UI interaction from ExecutePackageDownload

This could be resolved by running pmVmMock.Object.ExecutePackageDownload(id, pkgVer, ""); on the test thread but adding a DispatcherUtil.DoEvents() after the thread.Sleep bellow (would have been needed if there was UI interaction in ExecutePackageDownload)

Reply:
could you also make this test async, await the download and get rid of the sleep?

Reply:
It is not that easy to await the download. The `ExecutePackageDownload` would have to be refactored to return a task fo the download and install operations.